XCast/XCastSystemRemoteObject.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+* If not stated otherwise in this file or this component's LICENSE
+* file the following copyright and licenses apply:
+*
+* Copyright 2019 RDK Management
+* Copyright 2021 Liberty Global Service B.V.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+#include "XCastSystemRemoteObject.h"
+
+#include "utils.h"
+
+using namespace WPEFramework;
+
+rtDefineMethod(XCastSystemRemoteObject, systemRequest);
+
+rtDefineObject(XCastSystemRemoteObject, rtObject);
+
+void XCastSystemRemoteObject::registerRemoteObject(rtRemoteEnvironment* env) {
+    rtRemoteRegisterObject(env, XCAST_SYSTEM_OBJECT_SERVICE_NAME, this);
+}
+
+void XCastSystemRemoteObject::setService(RtNotifier * service){
+    m_observer = service;
```

Comment:
Is it possible that there will be race between setService and systemRequest at some point?

Reply:
in XCast::Initialize, they are called in sequence:
	(...)
    _rtConnector->setService(this)
    	---> first, RtXcastConnector::setService calls m_xcast_system_remote_object.setService
    if (XCast::isCastEnabled)
    {
        //TODO add rt intialization.
        if( _rtConnector->initialize())
        	---> then, RtXcastConnector::initialize() calls m_xcast_system_remote_object.registerRemoteObject
    (...)    
    
    - so I don't think so (before m_xcast_system_remote_object.registerRemoteObject is called, we should not be getting systemRequest calls)
    
    though, now that I look at it, some deinit code might actually be missing in XCast::Deinitialize